README.md

```diff
@@ -40,6 +40,161 @@ Then require the CSS in your `application.css` file:
+in place of `form_with`. Here's an example:
+
+```erb
+<%= bootstrap_form_with(model: @user, local: true) do |f| %>
+  <%= f.email_field :email %>
+  <%= f.password_field :password %>
+  <%= f.check_box :remember_me %>
+  <%= f.submit "Log In" %>
+<% end %>
+```
+
+This generates:
+
+```html
+<form role="form" action="/users" accept-charset="UTF-8" method="post">
+  <input name="utf8" type="hidden" value="&#x2713;" />
+  <div class="form-group">
+    <label class="form-control-label required">Email</label>
+    <input class="form-control" type="email" value="steve@example.com" name="user[email]" />
+  </div>
+  <div class="form-group">
+    <label class="form-control-label">Password</label>
+    <input class="form-control" type="password" name="user[password]" />
+    <span class="form-text text-muted">A good password should be at least six characters long</span>
+  </div>
+  <div class="form-check">
+    <label class="form-check-label">
+      <input name="user[remember_me]" type="hidden" value="0" />
+      <input class="form-check-input" type="checkbox" value="1" name="user[remember_me]" /> Remember menu
+    </label>
+  </div>
+  <input type="submit" name="commit" value="Log In" class="btn btn-secondary" data-disable-with="Log In" />
+</form>
+```
+
+If your form is not backed by a model, use `bootstrap_form_with` like this:
+
+```erb
+<%= bootstrap_form_with url: '/subscribe', local: true do |f| %>
+  <%= f.email_field :email, value: 'name@example.com' %>
+  <%= f.submit %>
+<% end %>
+```
+
+```html
+<form role="form" action="/subscribe" accept-charset="UTF-8" method="post">
+  <input name="utf8" type="hidden" value="&#x2713;" />
+  <div class="form-group">
+    <label class="form-control-label">Email</label>
+    <input value="name@example.com" class="form-control" type="email" name="email" />
+  </div>
+  <input type="submit" name="commit" value="Save " class="btn btn-secondary" data-disable-with="Save " />
+</form>
+```
+
+#### Important Differences Between `form_with` and `form_for`
+
+Rails 5.1 introduced `form_with`,
+which unifies the functionality previously found in `form_for` and `form_tag`.
+`form_for` and `form_tag` will be deprecated in a future version of Rails,
+so new applications should use `bootstrap_form_with`.
+
+`form_with` is different compared to `form_for` and `form_tag`.
+`bootstrap_form_width` basically just wraps `form_with`
+and adds some functionality,
+and so the different behaviour of `form_with`
+is refelected in `bootstrap_form_with`
+compared to `bootstrap_form_for`
+and `bootstrap_form_tag`.
+
+##### Ajax by Default
+`form_with` defaults to submitting forms via Javascript XHR calls,
+like `form_for` or `form_tag` would do if you specified `remote: true`.
+If you want the browser to submit the request
+the same way `form_for` and `form_tag` would do by default,
+you need to specify `local: true` as an option to `form_with`.
+
+##### No Default DOM IDs
```

Comment:
Thanks for catching this. I saw that the Rails PR had been merged, but in my rush to submit this PR, I didn't fix the README.

Reply:
I've fixed up the README. I've also written some test cases to make sure that `skip_default_ids: false` works as advertised, but I can't test them until I can run Rails 5.2. Will I be able to do that after #366 is merged, or do I need to figure out how to test against Rails 5.2 myself?

Reply:
You can just temporarily add `gem "rails", "5.2.0.beta3"` in Gemfile to test it

Reply:
Thanks, @GBH . I could only find beta2 on rubygems.org, but that's good enough for me for now.

Reply:
I need to slow down and read stuff instead of jumping to conclusions. The commit @GBH referenced (https://github.com/rails/rails/commit/d3893ec38ec61282c2598b01a298124356d6b35a) is not the commit I thought it was. I will have to rethink this. The commit makes Rails 5.2+ easier to work with, but I'll have to think about how to document Rails 5.1 and `form_with`.

Reply:
It looks like every `form_with` test may have to have a 5.1 and 5.2+ version. There's already too much duplication to test `form_with` and `form_for`. I'll think about this, but if anyone has any suggestions, I'd love to hear them.

tl;dr of the problem: `form_with` in Rails 5.1 doesn't generate any DOM IDs in the HTML. It looks like `form_with` in Rails 5.2 is going back to generating DOM IDs in the HTML by default.

Reply:
The thing is, you don't need to duplicate every single test there is. It's enough to make one for textfield and assume that all others work the same.